### PR TITLE
Add incremental processing for parsed events

### PR DESCRIPTION
**Pull request by kpodsiad**
Implements `getNextEvent` mentioned in https://github.com/VirtusLab/scala-yaml/issues/69

### CODE REVIEW

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/Parser.scala [hunk 1]
```diff
@@ -2,5 +2,7 @@ package org.virtuslab.yaml.internal.load.parse
 
 import org.virtuslab.yaml.YamlError
 import org.virtuslab.yaml.internal.load.reader.Tokenizer
+
 trait Parser:
-  def getEvents(in: Tokenizer): Either[YamlError, Seq[Event]]
+  def getEvents(): Either[YamlError, Seq[Event]]
```
Comment (lwronski) [inline yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/Parser.scala:7]: This means I can call getEvents and then getNextEvent. You should break it down into two traits.

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala [hunk 2]
```diff
@@ -49,202 +51,203 @@ private enum Production:
...
+        case Right(event) =>
+          if event != Event.StreamEnd then loop(event :: acc)
+          else Right((event :: acc).reverse)
         case Left(err) => Left(err)
     }
-    loop(Nil, List(ParseStreamStart))
+    loop(Nil)
 
-  type EventResult = Either[YamlError, (Event, List[Production])]
+  override def getNextEvent(): Either[YamlError, Event] =
```
Comment (lwronski) [inline yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala:72]: In any case, after the StreamEvent event, the parser will not return another event? I think we should use Option.

Comment (kpodsiad) [reply yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala:70]: I changed this behavior. Now, it's returning right with `Event.StreamEnd` value if there are no more events

**Review by lwronski — APPROVED** [review]
LGTM

**Pull request merged**